### PR TITLE
Add tests.  pull some inner classes out as package private classes.

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/api/IJobPersistence.java [hunk 1]
```diff
@@ -34,20 +35,16 @@
...
 	 * Chunk should be stored with a status of {@link ca.uhn.fhir.batch2.model.StatusEnum#QUEUED}
 	 *
-	 * @param theJobDefinitionId      The job definition ID
-	 * @param theJobDefinitionVersion The job definition version
-	 * @param theTargetStepId         The step ID that will be responsible for consuming this chunk
-	 * @param theInstanceId           The instance ID associated with this chunk
-	 * @param theDataSerialized       The data. This will be in the form of a map where the values may be strings, lists, and other maps (i.e. JSON)
-	 * @return Returns a globally unique identifier for this chunk. This should be a sequentially generated ID, a UUID, or something like that which is guaranteed to never overlap across jobs or instances.
+	 * @param theBatchWorkChunk the batch work chunk to be stored
+	 * @return Returns a globally unique identifier for this chunk. This should be a sequentially generated ID, a UUID, or something like that which is guaranteed to never overlap across jobs or instances.@return
```
Comment (TipzCM) [inline hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/api/IJobPersistence.java:39]: extra @return on the end here it looks like

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/impl/BatchWorkChunk.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+public class BatchWorkChunk {
+
+	public final String jobDefinitionId;
+	public final int jobDefinitionVersion;
+	public final String targetStepId;
+	public final String instanceId;
+	public final int sequence;
+	public final String serializedData;
```
Comment (TipzCM) [inline hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/impl/BatchWorkChunk.java:13]: Should these be.... my(whatevers)?

**Review by TipzCM — APPROVED** [review]
Does this require a changeLog?

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/impl/BatchWorkChunk.java (hunk 2 above)
Comment (fil512) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/impl/BatchWorkChunk.java:13]: not for public finals

**Pull request merged**